react-components/src/components/RuleBasedOutputs/utils.ts

```diff
@@ -494,9 +466,11 @@ const applyNodeStyles = (
   };
 
   const nodeIndexSet = ruleOutputAndStyleIndex.styleIndex.getIndexSet();
-  treeNodes.forEach((node) => {
-    nodeIndexSet.addRange(node.subtreeRange);
+  nodeIndexSet.clear();
+  treeNodes?.forEach((node) => {
+    nodeIndexSet.add(node.treeIndex);
   });
```

Comment:
I was talking to @pramodcog and he recommended me to use the single tree index in this case as we will coloring the parent of the descendant nodes which is the actual node that is linked to the asset and it will not require an extra fetch from the `model.getSubtreeTreeIndices` which gets 429 mostly and slow down the process.
Do you think we should reconsider that?

Reply:
moved to addRange again with a way better way now :D Thanks!

https://github.com/cognitedata/reveal/pull/4652/commits/01d9e4192802dde46298958dce3453c03f561cce